src/main/scala/com/fulcrumgenomics/commons/reflect/ReflectionUtil.scala

```diff
@@ -434,4 +433,48 @@ object ReflectionUtil {
     }
     enumConstants
   }
+
+  /** Returns the direct sub-classes of the given trait.  The sub-classes must be case objects.
+    *
+    * The order of the sub-classes returned is determined as follows.  First, search the companion of the trait for a
+    * `values` or `findValues` method, that returns the values in the desired order for the sealed trait hierarchy.
+    * Otherwise the direct sub-classes of the trait are returned.
+    */
+  def sealedTraitOptions[T](clazz: Class[T])(implicit evidence: ClassTag[T]): Try[Seq[T]] = Try {
+    val symbol = mirror.classSymbol(clazz)
+
+    // double-check it's a sealed trait!
+    require(symbol.isTrait && symbol.isSealed, f"Class '${clazz.getName}' is not a sealed trait")
+
+    // Get the companion
+    val companion = Try {
+      val companionObject = symbol.companion.asModule
+      val companionMirror = mirror.reflectModule(companionObject)
+      companionMirror.instance
+    }
+
+    // Get the values from the companion if possible
+    val companionValues: Option[Seq[Any]] = companion match {
+      case Failure(_) => None
+      case Success(_companion) =>
+        Iterator("values", "findValues").map { name =>
+          Try { _companion.getClass.getMethod(name).invoke(_companion).asInstanceOf[Seq[Any]] } }
+          .map {
+            case Success(_values) => _values
+            case Failure(_)       => Seq.empty[Any]
+          }
+          .find(_.nonEmpty)
+    }
+
+    companionValues.map { values => copy(values.toArray, clazz).toSeq }.getOrElse {
+      // Default to the direct sub-classes of the trait
+      symbol.knownDirectSubclasses.map { subSymbol =>
```

Comment:
Would it be hard to support indirect subclasses?  Ideally it would find all non-abstract subclasses I think.  I have at least one case somewhere where I have a sealed trait, with sub-traits and then a bunch of case objects extending the sub-traits.

Reply:
Sounds good.